parser/src/parser/tests/mod.rs

```diff
@@ -1,18 +1,619 @@
+        ])
+    }
+}
+
+macro_rules! constant {
+    ($name:ident = $value:literal) => {
+        Constant::new(
+            SourceSpan::UNKNOWN,
+            ident!($name),
+            ConstantExpr::Scalar($value),
+        )
+    };
+
+    ($name:ident = [$($value:literal),+]) => {
+        Constant::new(SourceSpan::UNKNOWN, ident!($name), ConstantExpr::Vector(vec![$($value),+]))
+    };
+
+    ($name:ident = [$([$($value:literal),+]),+]) => {
+        Constant::new(SourceSpan::UNKNOWN, ident!($name), ConstantExpr::Matrix(vec![$(vec![$($value),+]),+]))
+    };
+}
+
+macro_rules! vector {
+    ($($value:expr),*) => {
+        Expr::Vector(miden_diagnostics::Span::new(miden_diagnostics::SourceSpan::UNKNOWN, vec![$($value),*]))
+    }
+}
+
+macro_rules! matrix {
+    ($([$($value:expr),+]),+) => {
+        Expr::Matrix(miden_diagnostics::Span::new(miden_diagnostics::SourceSpan::UNKNOWN, vec![$(vec![$($value),+]),+]))
+    };
+}
+
+macro_rules! let_ {
+    ($name:ident = $value:expr => $($body:expr),+) => {
+        Statement::Let(Let::new(miden_diagnostics::SourceSpan::UNKNOWN, ident!($name), $value, vec![$($body),+]))
+    };
+}
+
+macro_rules! enforce {
+    ($expr:expr) => {
+        Statement::Enforce($expr)
+    };
+}
+
+macro_rules! enforce_all {
+    ($expr:expr) => {
+        Statement::EnforceAll($expr)
+    };
+}
+
+macro_rules! lc {
+    (($(($binding:ident, $iterable:expr)),+) => $body:expr) => {{
+        let context = vec![
+            $(
+                (ident!($binding), $iterable)
+            ),+
+        ];
+        ListComprehension::new(miden_diagnostics::SourceSpan::UNKNOWN, $body, context, None)
+    }};
+
+    (($(($binding:literal, $iterable:expr)),+) => $body:expr) => {{
+        let context = vec![
+            $(
+                (ident!($binding), $iterable)
+            ),+
+        ];
+        ListComprehension::new(miden_diagnostics::SourceSpan::UNKNOWN, $body, context, None)
+    }};
+
+    (($(($binding:ident, $iterable:expr)),*) => $body:expr, when $selector:expr) => {{
+        let context = vec![
+            $(
+                (ident!($binding), $iterable)
+            ),+
+        ];
+        ListComprehension::new(miden_diagnostics::SourceSpan::UNKNOWN, $body, context, Some($selector))
+    }};
+
+    (($(($binding:literal, $iterable:expr)),*) => $body:expr, when $selector:expr) => {{
+        let context = vec![
+            $(
+                (ident!($binding), $iterable)
+            ),+
+        ];
+        ListComprehension::new(miden_diagnostics::SourceSpan::UNKNOWN, $body, context, Some($selector))
+    }};
+}
+
+macro_rules! range {
+    ($range:expr) => {
+        Expr::Range(Span::new(SourceSpan::UNKNOWN, $range))
+    };
+}
+
+macro_rules! band {
+    ($lhs:expr, $rhs:expr) => {
+        mul!($lhs, $rhs)
+    };
+}
+
+macro_rules! bor {
+    ($lhs:expr, $rhs:expr) => {{
+        sub!(add!($lhs, $rhs), mul!($lhs, $rhs))
+    }};
+}
```

Comment:
I'd prefer `and` and `or`, and since we only have one type I don't think we risk lack of clarity